Selenium_Basics/Selenium_Basics/Selenium_Basics_2/EpamPageInteractionTests.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Specialized;
+
+namespace Selenium_Basics.Selenium_Basics_2
+{
+    public class EpamPageInteractionTests
+    {
+        private Actions _action;
+        private IWebDriver _driver;
+        private const string _searchIconButtonLocator = "//span [contains(@class, 'search-icon')]";
+        private const string _searchFormFieldLocator = "//*[@id = 'new_form_search']";
+        private const string _findButtonLocator = "//span [contains (@class, 'bth')]";
+
+
+        [SetUp]
+        public void Setup()
+        {
+            _driver = new ChromeDriver();
+            _action = new Actions(_driver);
+            _driver.Manage().Window.Maximize();
+            _driver.Navigate().GoToUrl("https://www.epam.com/");
+        }
+
+        [Test]
+        public void CheckAvailableCountriesIsPresentOnJobListingsPage()
+        {
+            _driver.FindElement(By.XPath("//*[contains(@class, 'item-link') and @href = '/careers']")).Click();
+            _driver.FindElement(By.XPath("//*[@class = 'arrow']//ancestor::*[contains (@href, 'job-listings') and @tabindex = '0']")).Click();
+            _action.ScrollToElement(_driver.FindElement(By.XPath("//*[contains (@href, 'net-developer-middle')]"))).Build().Perform();
+
+            var findCountries = _driver.FindElements(By.XPath("//*[contains (@class, 'js-tabs-title')]"));
+            var actualCountries = findCountries.Select(f => f.GetDomProperty("innerText").ToList());
+
+            List<string> expectedCountries = new() { "AMERICAS", "EMEA", "APAC" };
+
+            Assert.That(actualCountries, Is.EqualTo(expectedCountries), "Countries are not equal");
+        }
+
+        [TestCase ("Automation")]
+        [TestCase("Business Analysis")]
+        public void CheckSearchPanelResultIsCorrect(string ExpectedSearchResult)
+        {
+            _driver.FindElement(By.XPath(_searchIconButtonLocator)).Click();
+            _driver.FindElement(By.XPath(_searchFormFieldLocator)).SendKeys(ExpectedSearchResult);
+            _driver.FindElement(By.XPath(_findButtonLocator)).Click();
+
+            var SearchForm = _driver.FindElement(By.XPath("//input[contains (@aria-describedby, 'results')]"));
+
+            Assert.That(SearchForm.GetDomProperty("value"), Is.EqualTo(ExpectedSearchResult), "Search results don't match");
+        }
+
+        [TestCase (5, "Automation")]
+        public void CheckEnteredTextIsPresentInSearchResults(int expectedNumberOfWordToSearch, string searchQuery)
+        {
+            _driver.FindElement(By.XPath(_searchIconButtonLocator)).Click();
+            _driver.FindElement(By.XPath(_searchFormFieldLocator)).SendKeys(searchQuery);
+            _driver.FindElement(By.XPath(_findButtonLocator)).Click();
+            _driver.FindElement(By.XPath("//input[contains (@aria-describedby, 'results')]"));
+
+            var searchResults = _driver.FindElements(By.XPath("//div[@class = 'search-results__items']/article")).ToList();
```

Comment:
Nice way to check that AT LEAST 5 articles contain expected value. BUT! Please review logic in case we need to check that fisrt results should contain expected word. This code will check that any 5 artcles contain expected words, not 5 first ones